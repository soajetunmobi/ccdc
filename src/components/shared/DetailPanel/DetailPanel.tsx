import React, { FC } from 'react'

interface IDetailPanelProps {
  detail: RecordDetailObject
  onCloseHandler: CallableFunction
}

export const DetailPanel: FC<IDetailPanelProps> = props => {
  const { detail, onCloseHandler } = props

  const formatSubjectsToString = (subjects: Array<SubjectsType>): string => {
    const strVal: Array<string> = []
    subjects.forEach(sub => {
      strVal.push(sub.subject)
    })
    return strVal.join(',')
  }

  return (
    <div>
      <h2>Detail</h2>
      <div className='detail-row'>
        <div tabIndex={0} className='detail-label'>
          Publication year:
        </div>
        <div>{detail?.publicationYear}</div>
      </div>
      <div className='detail-row'>
        <div tabIndex={1} className='detail-label'>
          Publisher:
        </div>
        <div>{detail?.publisher}</div>
      </div>
      <div className='detail-row'>
        <div tabIndex={2} className='detail-label'>
          Subjects:
        </div>
        <div>{formatSubjectsToString(detail?.subjects)}</div>
      </div>
      <div className='detail-row'>
        <div tabIndex={3} className='detail-label'>
          URL:
        </div>
        <div>{detail?.url}</div>
      </div>
      <div className='margin-top--20'>
        <span
          role='button'
          tabIndex={4}
          data-testid='back-button'
          className='button'
          onClick={() => onCloseHandler()}
        >
          Back
        </span>
      </div>
    </div>
  )
}
