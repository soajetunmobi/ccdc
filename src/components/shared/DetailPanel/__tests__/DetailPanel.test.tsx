import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { DetailPanel } from '../DetailPanel'

describe('DetailPanel', () => {
  const detail: RecordDetailObject = {
    id: 'test-id',
    title: 'Test title',
    url: 'https://test.com/test',
    subjects: [{ subject: 'Test subject' }],
    publisher: 'Test publisher',
    publicationYear: 2021
  }
  const mockedOnCloseHandler = jest.fn()

  test('it renders correctly', () => {
    const { container } = render(
      <DetailPanel detail={detail} onCloseHandler={mockedOnCloseHandler} />
    )
    expect(container).toMatchSnapshot()
  })

  test('it renders publisher', () => {
    render(<DetailPanel detail={detail} onCloseHandler={mockedOnCloseHandler} />)
    expect(screen.getByText(detail.publisher)).toBeInTheDocument()
  })

  test('it renders publication year', () => {
    render(<DetailPanel detail={detail} onCloseHandler={mockedOnCloseHandler} />)
    expect(screen.getByText(detail.publicationYear)).toBeInTheDocument()
  })

  test('it renders url', () => {
    render(<DetailPanel detail={detail} onCloseHandler={mockedOnCloseHandler} />)
    expect(screen.getByText(detail.url)).toBeInTheDocument()
  })

  test('should go back to list', () => {
    render(<DetailPanel detail={detail} onCloseHandler={mockedOnCloseHandler} />)
    const backButton = screen.getByTestId('back-button')
    expect(backButton).toBeInTheDocument()
    fireEvent.click(backButton)
    expect(mockedOnCloseHandler).toBeCalled()
  })
})
