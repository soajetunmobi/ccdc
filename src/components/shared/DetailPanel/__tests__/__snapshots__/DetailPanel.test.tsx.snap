// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPanel it renders correctly 1`] = `
<div>
  <div>
    <h2>
      Detail
    </h2>
    <div
      class="detail-row"
    >
      <div
        class="detail-label"
        tabindex="0"
      >
        Publication year:
      </div>
      <div>
        2021
      </div>
    </div>
    <div
      class="detail-row"
    >
      <div
        class="detail-label"
        tabindex="1"
      >
        Publisher:
      </div>
      <div>
        Test publisher
      </div>
    </div>
    <div
      class="detail-row"
    >
      <div
        class="detail-label"
        tabindex="2"
      >
        Subjects:
      </div>
      <div>
        Test subject
      </div>
    </div>
    <div
      class="detail-row"
    >
      <div
        class="detail-label"
        tabindex="3"
      >
        URL:
      </div>
      <div>
        https://test.com/test
      </div>
    </div>
    <div
      class="margin-top--20"
    >
      <span
        class="button"
        data-testid="back-button"
        role="button"
        tabindex="4"
      >
        Back
      </span>
    </div>
  </div>
</div>
`;
